client/App.vue

```diff
@@ -121,35 +114,44 @@ export default {
       this.announcement.show = false;
       this.onVersionAnnouncementClose();
     },
-    async verifyNamespaceAccess() {
-      if (!this.$route.params.namespace) {
-        return;
-      }
-
-      const { routingConfig } = await this.$http(`/api/web-settings`);
-      if (!routingConfig?.defaultToNamespace) {
-        return;
-      }
-
-      const { defaultToNamespace } = routingConfig;
-
-      if (defaultToNamespace !== this.$route.params.namespace) {
-        this.onNotification({
-          message: `No access to namespace ${this.$route.params.namespace}. Redirecting to ${defaultToNamespace}`,
-          type: NOTIFICATION_TYPE_ERROR,
-        });
-        this.$router.push(`/namespaces/${defaultToNamespace}/workflows`);
-      }
+    async getWebSettings() {
+      this.webSettings = await this.$http(`/api/web-settings`);
     },
-    async verifyAuth() {
+    async getCurrentUser() {
       const me = await this.$http('/api/me');
-      if (me.isAuthEnabled) {
-        if (!me.user) {
+      this.currentUser = me?.user;
+    },
+    redirectIfApplicable() {
+      const { auth, routing } = this.webSettings;
+
+      if (auth?.enabled) {
+        if (!this.currentUser) {
           this.$router.push('/signin');
-        } else {
-          this.currentUser = me.user;
+          return;
         }
       }
+      if (routing?.defaultToNamespace) {
+        const { defaultToNamespace } = routing;
+        if (defaultToNamespace !== this.$route.params.namespace) {
+          this.onNotification({
+            message: `No access to namespace ${this.$route.params.namespace}. Redirecting to ${defaultToNamespace}`,
+            type: NOTIFICATION_TYPE_ERROR,
+          });
+          this.$router.push(`/namespaces/${defaultToNamespace}/workflows`);
+          return;
+        }
+      }
+    },
+    async announceNewVersionIfExists() {
+      const { show, message, link, version } = await getNewVersionAnnouncement(
+        this.$http,
+        this.onNotification
```

Comment:
nit: trailing comma

Reply:
this is not currently in the proj's prettier config